great_expectations_provider/example_dags/example_great_expectations_dag.py

```diff
@@ -3,113 +3,208 @@
+    }
 )
 
-# This runs an expectation suite against a data asset that passes the tests
-ge_batch_kwargs_list_pass = GreatExpectationsOperator(
-    task_id="ge_batch_kwargs_list_pass",
-    assets_to_validate=[
-        {
-            "batch_kwargs": {"path": data_file, "datasource": "data__dir"},
-            "expectation_suite_name": "taxi.demo",
-        }
-    ],
-    data_context_root_dir=ge_root_dir,
-    dag=dag,
+checkpoint_config = CheckpointConfig(
+    **{
+        "name": "taxi.pass.chk",
+        "config_version": 1.0,
+        "template_name": None,
+        "module_name": "great_expectations.checkpoint",
+        "class_name": "Checkpoint",
+        "run_name_template": "%Y%m%d-%H%M%S-my-run-name-template",
+        "expectation_suite_name": "taxi.demo",
+        "batch_request": None,
+        "action_list": [
+            {
+                "name": "store_validation_result",
+                "action": {"class_name": "StoreValidationResultAction"},
+            },
+            {
+                "name": "store_evaluation_params",
+                "action": {"class_name": "StoreEvaluationParametersAction"},
+            },
+            {
+                "name": "update_data_docs",
+                "action": {"class_name": "UpdateDataDocsAction", "site_names": []},
+            },
+        ],
+        "evaluation_parameters": {},
+        "runtime_configuration": {},
+        "validations": [
+            {
+                "batch_request": {
+                    "datasource_name": "my_datasource",
+                    "data_connector_name": "default_inferred_data_connector_name",
+                    "data_asset_name": "yellow_tripdata_sample_2019-01.csv",
+                    "data_connector_query": {"index": -1},
+                },
+            }
+        ],
+        "profilers": [],
+        "ge_cloud_id": None,
+        "expectation_suite_ge_cloud_id": None,
+    }
 )
 
-# This runs a checkpoint that will pass. Make sure the checkpoint yml file has the correct path to the data file.
-ge_checkpoint_pass = GreatExpectationsOperator(
-    task_id="ge_checkpoint_pass",
-    run_name="ge_airflow_run",
-    checkpoint_name="taxi.pass.chk",
-    data_context_root_dir=ge_root_dir,
-    dag=dag,
+passing_batch_request = BatchRequest(
+    **{
+        "datasource_name": "my_datasource",
+        "data_connector_name": "default_inferred_data_connector_name",
+        "data_asset_name": "yellow_tripdata_sample_2019-01.csv",
+        "data_connector_query": {"index": -1},
+    }
 )
 
-# This runs a checkpoint that will fail. Make sure the checkpoint yml file has the correct path to the data file.
-ge_checkpoint_fail = GreatExpectationsOperator(
-    task_id="ge_checkpoint_fail",
-    run_name="ge_airflow_run",
-    checkpoint_name="taxi.fail.chk",
+ge_data_context_root_dir_with_checkpoint_name_pass = GreatExpectationsOperator(
+    task_id="ge_data_context_root_dir_with_checkpoint_name_pass",
     data_context_root_dir=ge_root_dir,
```

Comment:
A good time to use `default_args` when defining the DAG object is if every Operator is going to use `data_context_root_dir`, then the value is passed to ever operator for you.